### ISSUE TITLE
RFSoC ZCU208 image

### ISSUE THREAD

**Issue by jswoboda** [casper-astro/tutorials_devel#78]
I'm looking for the CASPER image for the ZCU208. There isn't a full image available in the ZCU208 link on the [site](https://casper.groups.et.byu.net/). All that's there is folder called forjacob with a boot folder and a file named current.fsa. Are we supposed to use the ZCU216 image and do some sort of augmentation in the boot folder?

**Comment by mitchburnett** [comment]
We have not had dedicated access to a ZCU208 board for testing the CASPER image. We were working with someone who offered their board for some testing to try and get a CASPER image working. However, we never did end up getting anything tested board they ended up not getting to work on it as much as they wanted.

I have updated that ZCU208 directory with what was to be tested at the time. You are welcome to try it out. There may be some issue that need to be resolved. Without access to the platform we would be continuing to look for someone looking to help finish testing to support this platform.

**Comment by jswoboda** [comment]
I have a platform at Haystack if you would like some help testing let me know.

**Comment by mitchburnett** [comment]
The next step would be to take the new image placed on the CASPER RFSoC image [site](https://casper.groups.et.byu.net/) and see if it boots and see if the platform is responsive (either over serial or ssh). There is a new ZCU208 prebuilt bitstream for the RFDC tutorial also that is included in #79 that is shortly to be merged. That can also be tested if boot is then successful.

**Comment by jswoboda** [comment]
Hi @mitchburnett I've tried both the RFSoC ZCU208 images with both the file and fsa. They are both missing the contents of the /home/casper/bin directory that are in the the other systems. When I try to connect to it via ssh it works fine but if I try the casperfpga tools it fails to connect. Is there anyway I can get these files on the board? Do I need to re-flash the SD card?

**Comment by mitchburnett** [comment]
@jswoboda, this would be that I took one step too far in telling you that you can then run the tutorial after boots. Getting it to boot is a huge step though! Because, as mentioned earlier, access to a ZCU208 has been limited and so compiling and testing those other executables hasn't been possible. It looks like we hadn't even tried to run anything on the board. Just needed to get it to boot first.

I have added back to that ftp site for that zcu208 image a directory `zcu208update/` that has `tcpborphserver`, a new library to install, and a short readme to go about doing those steps. `tcpborphserver` packs into it most of the functionality that the other executables run. With the exception of starting the 8a34001 clock for the 100G core. This version of `tcpborphserver` hasn't been tested on a zcu208. Because the 208 has the same FPGA as the 4x2, a lot of the functionality will exist. Where I expect to run into trouble will be in communicating with the LMK/LMX and other off chip peripheral's. Much of that should be similar to the ZCU216. Though, it is a matter of getting it right.

So after you get these files updated and connected to the platform via casperfpga, we can then test running some programming of the lmk/lmx through casperfpga. This is covered in the RFDC interface tutorial, where `progpll` is called.

I think I may have a way to get access to a `zcu208` to also do some of my own testing to finish this out. Particularly now since you have an interest to use the platform. If I can't though, thanks for continuing to try and help us get this up and running!

**Comment by jswoboda** [comment]
@mitchburnett thanks for getting back to me. I updated the files you suggested and restarted the board but I was not able to get it to communicate through the casperfpga. 

I also checked `/lib/firmware` and the directory was empty. How would I get at those LMK/LMX clock files?

**Comment by mitchburnett** [comment]
@jswoboda, the CLK104 board is the same between the ZCU208 and ZCU216. I want to remember when I looked at ZCU208 that the CLK104 ended up connecting to the ZCU208 the same as the ZCU216. Meaning, that the same LMK/LMX clock files for the ZCU216 could also be used for the ZCU208 to make some initial tests. Then, from there they can be adjusted to suit your platform clocking needs based on the sample rates and requirements. This is done using TICS pro, a piece of TI software to generate the register files that are read programed to the PLL chips.

I just uploaded a clock file to that `zcu208update/` folder. This one provides a 250 MHz reference to the RFDC and a 125 MHz to the PL. That 125 MHz is then up'd to the 250 MHz fabric clock used to process samples. I want to remember in the RFDC tutorial I recently update it for the ZCU208, without being able to test it, to use these clock frequencies that this clock file produced. With the RFDC sampling at 4 GHz using a 250 MHz reference.
